streaming/util/shorthand.py

```diff
@@ -1,115 +1,371 @@
 # Copyright 2023 MosaicML Streaming authors
 # SPDX-License-Identifier: Apache-2.0
 
-"""Utilities for human-friendly argument shorthand."""
+"""Conversions between human-friendly string forms and int/float."""
```

Comment:
just a suggestion -- would the [humanize library](https://pypi.org/project/humanize/) be applicable for some of these functions? Would be nice to use an external library for stuff like this, removes some burden on us as well.

Reply:
Observations for posterity:
- the `humanfriendly` library is very pretty
- But it lacks "counts"
- So we are rolling our own functionality either way for that vertical
- Also, we are tight on time and this is enough, so I figure let's just go with this and revisit in 2024